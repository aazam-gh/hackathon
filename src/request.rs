use anyhow::Result;
use serde_json::json;

const API_URL: &str = "http://localhost:8000/graphql";

pub async fn graphql_request(query: &str) -> Result<serde_json::Value> {
    let client = reqwest::Client::new();

	let value = json!({
		"operationName": null,
		"variables": {},
		"query": query
	});

    let response = client.post(API_URL).json(&value).send().await?;

    Ok(response.json().await?)
}
