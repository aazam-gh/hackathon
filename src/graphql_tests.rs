use std::fs;

use anyhow::Result;
use insta::assert_json_snapshot;
use tokio::task::JoinSet;
use tracing::info;

use crate::request::graphql_request;

use super::ROOT_DIR;

pub async fn run_graphql_tests() -> Result<()> {
    info!("Run graphql assert tests");

    let tests_path = format!("{ROOT_DIR}/tests");

    let mut set: JoinSet<Result<()>> = JoinSet::new();

    for entry in fs::read_dir(tests_path)? {
        let path = entry?.path();

        if let Some(ext) = path.extension().and_then(|s| s.to_str()) {
            if ext == "graphql" {
                set.spawn(async {
                    tokio::spawn(async move {
                        let content = tokio::fs::read_to_string(&path).await?;

                        let response = graphql_request(&content).await?;
                        let name =
                            format!("request_{}", path.file_name().unwrap().to_string_lossy());

                        assert_json_snapshot!(name, response);

                        anyhow::Ok(())
                    })
                    .await?
                });
            }
        }
    }

    let results = set.join_all().await;

    info!("Execution of graphql tests finished");

    results.into_iter().collect()
}
