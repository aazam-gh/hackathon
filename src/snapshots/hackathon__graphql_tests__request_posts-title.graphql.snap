---
source: src/graphql_tests.rs
expression: response
---
{
  "data": {
    "posts": [
      {
        "title": "gaj peuk yuw vhrun euun zhprxx xytwg"
      },
      {
        "title": "atwtxz zbpg ipewla vldqvv chsqri sdikmts lpup ekri"
      },
      {
        "title": "pggij otg uwffzvu uxef eql toa bxsghi"
      },
      {
        "title": "balmi xykkkga qndf qoyhh aoxc"
      },
      {
        "title": "ihp sfoky cpwm xifcj ofypa"
      },
      {
        "title": "xex lstryah eep bhrndh gelxmp vjkz jar"
      },
      {
        "title": "debq qpju dvbsqpa faahaf cpfxfy ddyowd asec"
      },
      {
        "title": "mcpug dewi zgbal dgbadpj xcdwmls"
      },
      {
        "title": "qxsyiiw vmbak usgz ekfbzl mkfcbv"
      },
      {
        "title": "ffa grjgeb hhz wqyu xxree gqtxsss luziykb"
      },
      {
        "title": "vwwjhz hvzlwpe noiiad vmtnrfu yvn qkm vmu"
      },
      {
        "title": "dltps ryv fifsta tnslz islg"
      },
      {
        "title": "tyqbtlg nugr rjjbcja xyaag gahbpks bwuom knertw phqhkz"
      },
      {
        "title": "qoeaj zpry flsuhra jhwfkcn nrdh crjniwg hijpiw ndkmwz"
      },
      {
        "title": "jqrygw vljsz kyzrg vrcla gdi"
      },
      {
        "title": "jbs owkrzca iihkwww vvguj bkr sbkje"
      },
      {
        "title": "zew tpmmm avuiqah zrgfzn ltlb wnm opu"
      },
      {
        "title": "ohryfs onzktqs qdqgsrf ozprun gitnt uhz kpyciof"
      },
      {
        "title": "boupmv ronxkfk wwchahs mntmh ygqtt kjf ddjxpi"
      },
      {
        "title": "hagn fbrvrs jyyg kqm qurd hpd qiwheor"
      },
      {
        "title": "viivhak ygpcd zdokn kfpexxi dnoh oll"
      },
      {
        "title": "kopdfd yvxzzg ehvo pqjc vuvais hbswt"
      },
      {
        "title": "qkmw cjrtff mpdvikg caqls ozemdc qwb nduvtsb yik"
      },
      {
        "title": "yembpj sha gaob obgjqk vsj vqx ijwqf xopkz"
      },
      {
        "title": "cxyzdl scr rzlb zhexx uzrdwxk yfsmky osywqxp ntaio"
      },
      {
        "title": "zjetc zbgkxk gnlszzp plvff mvsglh epfll"
      },
      {
        "title": "qcmni zhf urgxyq omsaclj iqvd fltrp"
      },
      {
        "title": "mwanx hxzgnam idkas kumkk tzy qiels wjry"
      },
      {
        "title": "irrod siefi bays fzq lldi"
      },
      {
        "title": "hbkvfzg fnldq qvjoue urg lfxew"
      },
      {
        "title": "qvgtnzs txcx hocyvp txlf nyc"
      },
      {
        "title": "fgnj ioy niv ojxcvj iegr yyeofaz kgobey"
      },
      {
        "title": "vwaqvzz ohrxqn cagp pis guj cvvlv"
      },
      {
        "title": "csur vapvsw osiiuh tzujn aoifcj qkd"
      },
      {
        "title": "teuntnu jcj znzcbg lbdnz spzccuu qedlbie kcrxk"
      },
      {
        "title": "wiwqft fwcpuy kajnkd vilwmh kbqa mwqqzae lkxcj wbrn"
      },
      {
        "title": "ennzbkn yrgoev yxp ecusvzb vrlkz"
      },
      {
        "title": "glzcuvw eaouafc nrcyk eztjz hjox vfd ilf jxfreh"
      },
      {
        "title": "gtw yrrg mafuhud thl yccfc pwuaz ypgr kzzrz"
      },
      {
        "title": "qmivzy zqnjh erviq yrio uenhlt kivuql"
      },
      {
        "title": "ymch zivug aeni pun ndsihz glgdfs"
      },
      {
        "title": "frtm fro yjgjc zkny hrjmgmq vwtzqp yeq godz"
      },
      {
        "title": "xbgfiv srr anc gxcv yapdxj"
      },
      {
        "title": "ceplx neetyo hsgr wnut qksdcy ytkwcez zhe pspzpvo"
      },
      {
        "title": "klisfhi yojrtqn lutfcv rwn zmfc wrc bgjes"
      },
      {
        "title": "qpm npxaxh kxusb eufwbo zyffi jeoj"
      },
      {
        "title": "longqm aaiotw vkay wvb wwkxols"
      },
      {
        "title": "xbpwiz xtdgo elh hgbbgma iowp"
      },
      {
        "title": "kralft ruhdn oak llosdi fglhn hyhrk"
      },
      {
        "title": "sqgjd obe trtcghq vkhfrn uoswkpv ybfbo egwvdwy beax"
      },
      {
        "title": "squv ermz gaxcoa gunfuc xzmfl fvavze"
      },
      {
        "title": "mbjypgh udmzdrt yqxyp kqb agy emqsk"
      },
      {
        "title": "fxpovp zjfunh eoem shc xzc"
      },
      {
        "title": "yidpyma ohpms frlxhn pcpfsdp qblzuu gkasr ffwwxj"
      },
      {
        "title": "epwyf lmyt vjvmm lmodrn ovje gonlr lecgx"
      },
      {
        "title": "ydr kai bxnado kbgq ebrscf xlouwi"
      },
      {
        "title": "qtujv rii hola ylhkzoh eyuu"
      },
      {
        "title": "qrjdn nnkssq wklfbb siyu jhe wdflcmj aoshgry"
      },
      {
        "title": "yzjxyn xfptjvj yfhvco vxihp tnln zaooxha"
      },
      {
        "title": "wpiy ysn mxfn fnlb yjtz lyix eaabv"
      },
      {
        "title": "zwom phcbo ifxxgla nuhr ite pzyxuq org"
      },
      {
        "title": "ifuk ynue fbmiey fhsbzbh lyktgvw dpjmh camg"
      },
      {
        "title": "efqs nxweet tadh dkxbx szpyrkj bdi"
      },
      {
        "title": "guod tkk hjvz kxv dhjkc daijvnl"
      },
      {
        "title": "ctkhy hwstwn ohihivz qidvu ycwlko yed kid"
      },
      {
        "title": "wwgbzdu arlg lbeg tydbdlp zkxlb"
      },
      {
        "title": "mlyznz poea vppxdt tpdtiic ixendao myjkjt"
      },
      {
        "title": "cyqi nqs mpc vbnpsl kxxdtg dxoubcn kyflb"
      },
      {
        "title": "ewn rytnav zidaqqv jhgqhgd oocx lgj frqotdk"
      },
      {
        "title": "olrfd akrwh cstvssd xtgh stj ftscnk"
      },
      {
        "title": "xpjlfm imgsfzj ldsn kroh zfsn"
      },
      {
        "title": "wkod ger huyit pfu dtp"
      },
      {
        "title": "hnx iohsga rpz qmqau senq zdwug cqzzjt"
      },
      {
        "title": "lqtmwj xeeag awscc ytkioe kzr hevjao"
      },
      {
        "title": "ubzz fje blnkgqy cye ypaou kuoy hqhz cat"
      },
      {
        "title": "lxa hvkc uhef twsndzr oyk"
      },
      {
        "title": "njiyn uxqshs ttnmqb kssli occcnj aoc"
      },
      {
        "title": "ocgibjl xhh prtjjk wfg ssqsdw nxns irgxz ujdw"
      },
      {
        "title": "xmj jcmjj axgzgv hhqub ysixijr tntjfwk"
      },
      {
        "title": "fkz dprxo rixhz nsms muz ndpke biuij cxn"
      },
      {
        "title": "jzwu cdrqo plky wpnudz wgdnqyx yobu nlydbw"
      },
      {
        "title": "blly dhxq amuacz vkhe waxsqgm szbp"
      },
      {
        "title": "nfrwgp abpkjy mev bbmufo gtyfvd hjke ibs"
      },
      {
        "title": "aup ovnwcm pffebc wder aabdx xkshmq byi jeaz"
      },
      {
        "title": "bhlejm zth tirvx wfvna mjojpde oqvzo"
      },
      {
        "title": "dsv mdgiipp pcvi xuor wfjs idbk tlglegc"
      },
      {
        "title": "zkf noecln odd srj wfcommy lfngxpt jxfb"
      },
      {
        "title": "rfbxriw niqsehn lwwi niid moeecg shf dsui"
      },
      {
        "title": "ctyplf drjp svff zopvw txdtb"
      },
      {
        "title": "lwyhxb mvkgole jnw eqbnmm fxuxkym elrfmn"
      },
      {
        "title": "ucuxon glbszu sqmerr uxrgqsz ribpjm yulagt"
      },
      {
        "title": "fqbxy xpttxec qzndfgd wjjvw qyhjkuj mjyufl brvk"
      },
      {
        "title": "qhx ytgzoel lxy cxhtvp tskgy ysirbo"
      },
      {
        "title": "xzzvj dwfsvcb fzb qfld vcyc ityfhkv irnxug mttfp"
      },
      {
        "title": "xkxeea qjpmjg vrcqlz dahxeol aenga pczkaf eckr nfhdrzs"
      },
      {
        "title": "heyo xmugz lvw ehdf ppxassi"
      }
    ]
  }
}
